packages/run-protocol/src/vpool-xyk-amm/addPool.js

```diff
@@ -35,20 +41,30 @@ export const makeAddIssuer = (zcf, isInSecondaries, brandToLiquidityMint) => {
     const liquidityKeyword = `${keyword}Liquidity`;
     zcf.assertUniqueKeyword(liquidityKeyword);
 
-    return E.when(
-      zcf.makeZCFMint(
-        liquidityKeyword,
-        AssetKind.NAT,
-        harden({ decimalPlaces: 6 }),
-      ),
-      /** @param {ZCFMint} mint */
-      async mint => {
-        await zcf.saveIssuer(secondaryIssuer, keyword);
-        brandToLiquidityMint.init(secondaryBrand, mint);
-        const { issuer: liquidityIssuer } = mint.getIssuerRecord();
-        return liquidityIssuer;
-      },
+    const mint = await zcf.makeZCFMint(
+      liquidityKeyword,
+      AssetKind.NAT,
+      harden({ decimalPlaces: 6 }),
     );
+    await zcf.saveIssuer(secondaryIssuer, keyword);
+    const issuer = zcf.getIssuerForBrand(secondaryBrand);
+    console.log(
+      'Saved issuer',
+      secondaryIssuer,
+      'to keyword',
+      keyword,
+      'and got back',
+      issuer,
+    );
+    // this ensures that getSecondaryIssuer will return
+    brandToLiquidityMint.init(secondaryBrand, mint);
+    const { issuer: liquidityIssuer } = mint.getIssuerRecord();
+    // ??? still need this deferred lookup?
+    const addIssuerToReserve = getAddIssuerToReserve();
```

Comment:
yes, I believe so.  It also makes it simpler to transition to a governed parameter if we decide to go that way.